javascript/extractor/src/com/semmle/js/extractor/AutoBuild.java

```diff
@@ -1113,13 +1204,25 @@ private void doExtract(FileExtractor extractor, Path file, ExtractorState state)
 
     try {
       long start = logBeginProcess("Extracting " + file);
-      Integer loc = extractor.extract(f, state);
-      if (!extractor.getConfig().isExterns() && (loc == null || loc != 0)) seenCode = true;
+      ParseResultInfo loc = extractor.extract(f, state);
+      if (!extractor.getConfig().isExterns() && (loc == null || loc.getLinesOfCode() != 0)) seenCode = true;
       if (!extractor.getConfig().isExterns()) seenFiles = true;
+      for (ParseError err : loc.getParseErrors()) {
+        writeDiagnostics(err.getMessage(), JSDiagnosticKind.PARSE_ERROR);
```

Comment:
Shouldn't we include a location for parse error? I suppose the message string contains them, but it is better to have them logged as a separate field to the status page can create links to the files/lines.

Reply:
Sounds like a good idea.  
It turned out to require some re-writing, as the `ParseError` didn't have a file associated. 

(The file is unambiguous where parse-errors gets written to the DB).